tests/components/renault/test_scan_interval.py

```diff
@@ -0,0 +1,124 @@
+"""Tests for Renault scan interval."""
+
+from datetime import timedelta
+from unittest.mock import patch
+
+import pytest
+from renault_api.kamereon.exceptions import AccessDeniedException, NotSupportedException
+
+from homeassistant.components.renault.const import MAX_CALLS_PER_HOURS
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from .conftest import _get_fixtures, patch_get_vehicle_data
+
+pytestmark = pytest.mark.usefixtures("patch_renault_account", "patch_get_vehicles")
+
+
+@pytest.fixture(autouse=True)
+def override_platforms() -> None:
+    """Override PLATFORMS."""
+    with patch("homeassistant.components.renault.PLATFORMS", [Platform.SENSOR]):
+        yield
+
+
+@pytest.mark.parametrize(
+    ("vehicle_type", "expected_coordinator_count", "expected_scan_interval"),
+    [
+        # One vehicle with 5 coordinators
+        ("zoe_40", 5, timedelta(seconds=(3600 * 5) / MAX_CALLS_PER_HOURS)),
+        # One vehicle with partial coordinators
+        ("captur_fuel", 4, timedelta(seconds=(3600 * 4) / MAX_CALLS_PER_HOURS)),
+    ],
+)
+async def test_scan_interval_calculation_single_vehicle(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    device_registry: dr.DeviceRegistry,
+    vehicle_type: str,
+    expected_coordinator_count: int,
+    expected_scan_interval: timedelta,
+) -> None:
+    """Test the scan interval is properly calculated for a single vehicle."""
+    # Set up fixtures for the vehicle
+    mock_fixtures = _get_fixtures(vehicle_type)
+
+    with patch_get_vehicle_data() as patches:
+        # Set up the fixtures
+        for key, value in patches.items():
+            value.return_value = mock_fixtures[key]
+
+        # Setup the component
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+        # Verify that the vehicle has the expected number of coordinators
+        vehicle = list(config_entry.runtime_data.vehicles.values())[0]
+        assert len(vehicle.coordinators) == expected_coordinator_count
```

Comment:
It is bad practise to test the internals - it is better to use the freezer and check that the coordinators are called at the right time.